### PR TITLE
simple_switch: make the prio-queue a command line option

### PR DESCRIPTION
**Pull request by duskmoon314**
## Introduction

This PR solves issue #1065 

- Use `QueueingLogicPriRL` instead of `QueueingLogicRL` to make the priority queue a command-line option of simple_switch(_grpc)
- Add optional argument to `set_queue_depth` and `set_queue_rate` to set the depth/rate of a queue with a specific priority

### CODE REVIEW

File: targets/simple_switch/thrift/src/SimpleSwitch_server.cpp [hunk 1]
```diff
@@ -104,6 +104,14 @@ class SimpleSwitchHandler : virtual public SimpleSwitchIf {
     }
   }
 
+  int32_t set_egress_priority_queue_depth(const int32_t port_num,
+                                          const int32_t priority,
+                                          const int32_t depth_pkts) {
+    bm::Logger::get()->trace("set_egress_jpriority_queue_depth");
```
Comment (antoninbas) [inline targets/simple_switch/thrift/src/SimpleSwitch_server.cpp:110]: typo

**Review by antoninbas — COMMENTED** [review]
LGTM except for one typo in a log message.
Your TODOs do not need to be handled as part of this PR, and anyway I am not sure what you mean exactly by Mininet integration,

**Comment by duskmoon314** [comment]
The typo is fixed. Adjust some lines according to the ci.

"Mininet integration" means adding priority queues to the arguments of P4Switch's constructor. Thus we can use priority queues in the mininet.

**Review by antoninbas — COMMENTED** [review]
sorry I have one more request. Could you rename all `nb_queues` variables to `nb_queues_per_port` and `default_nb_queues` variables to `default_nb_queues_per_port`? Apart from that we are good to go. Thanks!

**Comment by duskmoon314** [comment]
Variables are renamed. I haven't checked if everything matches the style guide. (Don't know how to run cpplint locally, sorry 😢 )

File: targets/simple_switch/main.cpp [hunk 2]
```diff
@@ -46,6 +46,9 @@ main(int argc, char* argv[]) {
   simple_switch_parser.add_uint_option(
       "drop-port",
       "Choose drop port number (default is 511)");
+  simple_switch_parser.add_uint_option(
+      "priority-queues",
+      "Numbers of priority queues (default is 1)");
```
Comment (rst0git) [inline targets/simple_switch/main.cpp:51]: *Number of priority queues

File: targets/simple_switch/simple_switch.h [hunk 3]
```diff
@@ -79,14 +74,17 @@ class SimpleSwitch : public Switch {
...
  private:
   using clock = std::chrono::high_resolution_clock;
 
  public:
   // by default, swapping is off
   explicit SimpleSwitch(bool enable_swap = false,
-                        port_t drop_port = default_drop_port);
+                        port_t drop_port = default_drop_port,
+                        size_t nb_queues_per_port
+                            = default_nb_queues_per_port);
```
Comment (rst0git) [inline targets/simple_switch/simple_switch.h:87]: Would it be possible to have `size_t nb_queues_per_port = default_nb_queues_per_port` on a single line?

File: targets/simple_switch_grpc/main.cpp [hunk 4]
```diff
@@ -110,6 +110,9 @@ main(int argc, char* argv[]) {
       "dp-grpc-server-addr",
       "Use a gRPC channel to inject and receive dataplane packets; "
       "bind this gRPC server to given address, e.g. 0.0.0.0:50052");
+  simple_switch_parser.add_uint_option(
+    "priority-queues",
+    "Numbers of priority queues (default is 1)");
```
Comment (rst0git) [inline targets/simple_switch_grpc/main.cpp:115]: *Number of priority queues

File: targets/simple_switch/simple_switch.h (hunk 3 above)
Comment (duskmoon314) [reply targets/simple_switch/simple_switch.h:87]: I see the column of the semicolon is 81 in VSCode, so I put it on two lines. Is it better to put on a single line?

Comment (duskmoon314) [reply targets/simple_switch/simple_switch.h:87]: Oh, I made a mistake. It is on the 80th column. I will fix this.

**Comment by duskmoon314** [comment]
Adjust codes according to suggestions of @rst0git

File: targets/simple_switch/main.cpp [hunk 5]
```diff
@@ -46,6 +46,9 @@ main(int argc, char* argv[]) {
   simple_switch_parser.add_uint_option(
       "drop-port",
       "Choose drop port number (default is 511)");
+  simple_switch_parser.add_uint_option(
+      "priority-queues",
+      "Number ofpriority queues (default is 1)");
```
Comment (rst0git) [inline targets/simple_switch/main.cpp:51]: @duskmoon314 Could you add space between `of` and `priority`?

Comment (duskmoon314) [reply targets/simple_switch/main.cpp:51]: Ok, in a second

**Review by rst0git — APPROVED** [review]
LGTM

**Review by antoninbas — APPROVED** [review]

**Comment by antoninbas** [comment]
Thanks for contributing this @duskmoon314 !

**Pull request merged**